include/cuco/detail/bitwise_compare.cuh

```diff
@@ -73,5 +75,39 @@ __host__ __device__ bool bitwise_compare(T const& lhs, T const& rhs)
                                                           reinterpret_cast<char const*>(&rhs));
 }
 
+/**
+ * @brief Customization point that can be specialized to indicate that it is safe to perform bitwise
+ * equality comparisons on objects of type `T`.
+ *
+ * By default, only types where `std::has_unique_object_representations_v<T>` is true are safe for
+ * bitwise equality. However, this can be too restrictive for some types, e.g., floating point
+ * types.
+ *
+ * User-defined specializations of `is_bitwise_comparable` are allowed, but it is the users
+ * responsibility to ensure values do not occur that would lead to unexpected behavior. For example,
+ * if a `NaN` bit pattern were used as the empty sentinel value, it may not compare bitwise equal to
+ * other `NaN` bit patterns.
+ *
+ */
+template <typename T, typename = void>
+struct is_bitwise_comparable : std::false_type {
```

Comment:
This trait shouldn't be in `detail` since it is intended to be user facing.

Reply:
Which is better, moving `is_bitwise_comparable` to another header like `traits.cuh` or keeping it in the same file but moving to `cuco` namespace?

Reply:
I like `traits.cuh` in the main include directory